### PR TITLE
[tests-only] [full-ci] removing the setresponse in given/then step in publicwebdav context

### PR DESCRIPTION
**Pull request by KarunAtreya**
## Description
We have used setResponse() and $this->response in the Given/Then steps and some helper functions (maybe to reuse existing available methods). But storing responses from Given/Then steps and helper functions is not a good idea because it can lead to a false positive assertion in the Then steps.
So, check the use of setResponse() and $this->response in
- Given steps
- Then steps (Then steps can use $this->response but must prevent saving to it)
- Helper functions

So this pr make the above changes in `PublicWebdavContext`
## Related Issue
https://github.com/owncloud/ocis/issues/7082

## Motivation and Context
- To  remove setResponse() and $this->response in the Given/Then steps and some helper functions
- To avoid false positive assertions

## How Has This Been Tested?

- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/editPublicLinkOfSpace.feature [hunk 1]
```diff
@@ -43,7 +43,8 @@ Feature: A manager of the space can edit public link
       | displayname_owner | %displayname%         |
       | name              | <linkName>            |
     And the public should be able to download file "/test.txt" from inside the last public link shared folder using the new public WebDAV API with password "<password>"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/editPublicLinkOfSpace.feature:45]: do we need this step. IMO, steps below confirm it

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 2]
```diff
@@ -37,7 +37,8 @@ Feature: Share spaces via link
       | uid_owner         | %username%            |
       | name              | <linkName>            |
     And the public should be able to download file "/test.txt" from inside the last public link shared folder using the new public WebDAV API with password "<password>"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:39]: do we need this step. IMO, steps below confirm it

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature [hunk 3]
```diff
@@ -61,9 +61,11 @@ Feature: create a public link share
     And the public should be able to download the last publicly shared file using the new public WebDAV API with password "%public%" and the content should be "Random data"
     And the HTTP status code should be "200"
     And the public download of the last publicly shared file using the new public WebDAV API with password "%regular%" should fail with HTTP status code "401"
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:63]: this step should be covered by steps below:
```suggestion
     When the public downloads the last public link shared file with password "%regular%" using the new public WebDAV API
    Then the HTTP status code should be "401"
    And the value of the item "//s:message" in the response should match "/Username or password was incorrect/"
```

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature [hunk 4]
```diff
@@ -61,9 +61,11 @@ Feature: create a public link share
...
-    And the value of the item "//s:message" in the response should match "/Username or password was incorrect/"
+    When the public downloads the last public link shared file with password "%regular%" using the new public WebDAV API
+    Then the value of the item "//s:message" in the response should match "/Username or password was incorrect/"
     And the public download of the last publicly shared file using the new public WebDAV API without a password should fail with HTTP status code "401"
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:66]: same here

File: tests/acceptance/features/coreApiSharePublicLink2/reShareAsPublicLinkToSharesNewDav.feature [hunk 5]
```diff
@@ -38,7 +38,8 @@ Feature: reshare as public link
     Then the OCS status code should be "<ocs_status_code>"
     And the HTTP status code should be "200"
     And the public should be able to download file "file.txt" from inside the last public link shared folder using the new public WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink2/reShareAsPublicLinkToSharesNewDav.feature:40]: same as above (for others below as well)

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 6]
```diff
@@ -230,29 +232,47 @@ public function thePublicRenamesFileFromTheLastPublicShareUsingThePasswordPasswo
...
+	}
+
+	/**
+	 * @param string $path
+	 * @param string|null $password
+	 * @param string $publicWebDAVAPIVersion
+	 *
+	 * @return ResponseInterface
+	 */
+	public function downloadTheFileInsideThePublicSharedFolderWithPassword(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:270]:
```suggestion
	public function downloadFileFromPublicFolderWithPassword(
```

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 7]
```diff
@@ -270,30 +290,29 @@ public function publicDownloadsTheFileInsideThePublicSharedFolderWithPassword(
...
 	 * @param string $path
 	 * @param string $password
 	 * @param string $range ignored when empty
 	 * @param string $publicWebDAVAPIVersion
 	 *
-	 * @return void
+	 * @return ResponseInterface
 	 */
-	public function publicDownloadsTheFileInsideThePublicSharedFolderWithPasswordAndRange(
+	public function downloadTheFileInsideThePublicSharedFolder(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:310]:
```suggestion
	public function downloadFileFromPublicFolder(
```

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php (hunk 6 above)
Comment (saw-jan) [reply tests/acceptance/features/bootstrap/PublicWebDavContext.php:270]: Could this be removed and replaced by `downloadTheFileInsideThePublicSharedFolder`?

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature [hunk 8]
```diff
@@ -59,10 +59,11 @@ Feature: create a public link share
       | uid_owner              | %username%      |
       | name                   |                 |
     And the public should be able to download the last publicly shared file using the new public WebDAV API with password "%public%" and the content should be "Random data"
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:61]: This step hould not be necessary, right?
```suggestion
```

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature [hunk 9]
```diff
@@ -59,10 +59,11 @@ Feature: create a public link share
...
-    And the HTTP status code should be "200"
-    And the public download of the last publicly shared file using the new public WebDAV API with password "%regular%" should fail with HTTP status code "401"
+    When the public downloads the last public link shared file with password "%regular%" using the new public WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:62]: do we have step like this?
```suggestion
    When the public tries to download the last public link shared file with password "%regular%" using the new public WebDAV API
```

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 10]
```diff
@@ -230,34 +232,32 @@ public function thePublicRenamesFileFromTheLastPublicShareUsingThePasswordPasswo
...
 	 *
 	 * @return void
 	 */
-	public function publicDownloadsTheFileInsideThePublicSharedFolderWithPassword(
-		string $path,
-		?string $password = "",
-		string $publicWebDAVAPIVersion = "old"
-	):void {
-		$this->publicDownloadsTheFileInsideThePublicSharedFolderWithPasswordAndRange(
+	public function downloadPublicFileInsideAFolderWithPassword(string $path, string $publicWebDAVAPIVersion = "old", string $password = ""):void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:253]:
```suggestion
	public function publicDownloadsFileFromInsideLastPublicSharedFolderWithPassword(string $path, string $publicWebDAVAPIVersion = "old", string $password = ""):void {
```

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 11]
```diff
@@ -230,34 +232,32 @@ public function thePublicRenamesFileFromTheLastPublicShareUsingThePasswordPasswo
...
 			"",
 			"",
 			$publicWebDAVAPIVersion
 		);
+		$this->featureContext->setResponse($response);
 	}
 
 	/**
-	 * @When /^the public downloads file "([^"]*)" from inside the last public link shared folder with password "([^"]*)" using the (old|new) public WebDAV API$/
+	 * @When /^the public downloads file "([^"]*)" from inside the last public link shared folder using the (old|new) public WebDAV API with password "([^"]*)"$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:245]:
```suggestion
	 * @When /^the public downloads file "([^"]*)" from inside the last public link shared folder with password "([^"]*)" using the (old|new) public WebDAV API$/
```

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature (hunk 9 above)
Comment (KarunAtreya) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:62]: no we don't

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature (hunk 8 above)
Comment (KarunAtreya) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:61]: I think it is. In the scenario public link is created with password. So public download has three conditions:
- to be able to download with password
- download fail with wrong password
- download fail with no password

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature (hunk 9 above)
Comment (saw-jan) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:62]: so then we can have one (multiple steps can be assigned to a single stef def)

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature (hunk 8 above)
Comment (saw-jan) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:61]: Does that Then step save response?

Comment (KarunAtreya) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:61]: no it does not, now

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature (hunk 9 above)
Comment (KarunAtreya) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:62]: do you mean to make a new step

Comment (saw-jan) [reply tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:62]: Yep, we can link multiple steps like this:
```php
	/**
	 * @When /^the public downloads the last public link shared file with password "([^"]*)" using the (old|new) public WebDAV API$/
	 * @When /^the public tries to download the last public link shared file with password "([^"]*)" using the (old|new) public WebDAV API$/
	 *
```

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Pull request merged**